examples/nlp/Tweet-classification-using-TFDF.py

```diff
@@ -0,0 +1,272 @@
+
+dataset description :
+
+1. Files
+    - train.csv - the training set
+
+2. Columns
+    - id - a unique identifier for each tweet
+    - text - the text of the tweet
+    - location - the location the tweet was sent from (may be blank)
+    - keyword - a particular keyword from the tweet (may be blank)
+- target - in train.csv only, this denotes whether a tweet is about a real disaster (1)
+or not (0)
+"""
+
+# Turn .csv files into pandas DataFrame's
+df = pd.read_csv(
+    "https://raw.githubusercontent.com/IMvision12/Tweets-Classification-NLP/main/train.csv"
+)
+print(df.head())
+
+# Printing shape of the training dataset
+print(f"Training dataset shape: {df.shape}")
+
+# Shuffling training data
+df_shuffled = df.sample(frac=1, random_state=42)
+# Dropping id, keyword and location columns as these columns consists of mostly nan values
+# we will be using only text and target columns
+df_shuffled.drop(["id", "keyword", "location"], axis=1, inplace=True)
+df_shuffled.reset_index(inplace=True, drop=True)
+print(df_shuffled.head())
+
+print(df_shuffled.info())
+
+# Printing total number of Disaster and non-Disaster tweets
+print(
+    f"Total Number of disaster and non-disaster tweets\n{df_shuffled.target.value_counts()}"
+)
+
+# Viewing 5 records from training data
+for ind, counter in enumerate(df_shuffled.index):
+    print(f"Target : {df_shuffled['target'][ind]}")
+    print(f"Text : {df_shuffled['text'][ind]}")
+    if counter == 5:
+        break
+
+# Splitting dataset into train and test
+test_df = df_shuffled.sample(frac=0.1, random_state=42)
+train_df = df_shuffled.drop(test_df.index)
+
+print(f"Using {len(train_df)} samples for training and {len(test_df)} for validation")
+
+print(train_df["target"].value_counts())
+print(test_df["target"].value_counts())
+
+"""
+## Convert data to tf.Dataset
+"""
+
+
+def create_dataset(dataframe):
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (df["text"].to_numpy(), df["target"].to_numpy())
+    )
+    dataset = dataset.batch(100)
+    dataset = dataset.prefetch(tf.data.AUTOTUNE)
+    return dataset
+
+
+train_ds = create_dataset(train_df)
+test_ds = create_dataset(test_df)
+
+"""
+## Downloading Pre-trained Embeddings
+
+The Universal Sentence Encoder encodes text into high-dimensional vectors that can be
+used for text classification, semantic similarity, clustering and other natural language
+tasks.It is trained on a variety of data sources and a variety of tasks. The input is
+variable length English text and the output is a 512 dimensional vector.
+
+To understand better about pre-trained embeddings : 
+[Universal Sentence Encoder](https://tfhub.dev/google/universal-sentence-encoder/4)
+
+
+"""
+
+sentence_encoder_layer = hub.KerasLayer(
+    "https://tfhub.dev/google/universal-sentence-encoder/4"
+)
+
+"""
+## Build a model
+
+We will create 2 models In first model raw text will be directly passed to the Gradient
```

Comment:
This is reversed, right?

The "first model" (model_1) uses pre-trained embeddings, while the "second model" (model_2) consumes text natively.

Reply:
Yeah